client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
+2. **HTTP support**: while the GraphQL specification is independent from the transport layer, we believe that in practice it is relevant to support HTTP features such as timeouts and headers
+3. **Consistency with MicroProfile**:
+
+    * No dependency outside MicroProfile core 
+    * Configuration exclusively based on MicroProfile Config
+    * Support of JSON-B format directives
+    * As consistent as possible to JAX-RS client API 
+4. Baseline for a "high level" API
+5. Open to *reactive programming*
+
+
+== Non goals
+
+This specification doesn't aim at defining a high level GraphQL client API. However, it is intended to be used "behind the scene" by it.
+
+== Focus of release 1.1
+
+A first version of the client API is planned with MicroProfile GraphQL 1.1. For this first step, we propose to focus on the following core features.
+
+==== GraphQL support
+
+* Operations: queries and mutations
+* Partial results
+* Errors
+* Arguments
+* Variables
+* Interface
+
+==== HTTP supports
+
+* Headers: reading and writing
+* Connection timeout
+* Read timeout
+
+==== JSON-B support
+
+* Format directives on fields: dates...
+
+==== MicroProfile Config
+
+Configuration exclusively based on MicroProfile Config.
+
+==== Synchronous client API
+
+We propose a synchronous-only API for this first release. 
+
+A reactive API, similar to JAX-RS client API (see chapter 5.7) will be studied for a next release.
+
+==== Java 8 support
+
+Java 8 is still widely used in the industry and we propose to stick to it for a broader adoption.
+
+== Next steps
+
+To be studied for next releases:
+
+* GraphQL subscription, union
+* Reactive API
+* https://spec.graphql.org/draft/#sec-Document[Multiple operations inside a GraphQL document]
+
+== API building-blocks
+
+All classes described in this document are part of the package __org.eclipse.microprofile.graphql.client__.
+
+The API is based on 5 main classes:
+
+* **Builder**: prepares and builds GraphQL operations
+* **Request**: represents a read-to-run GraphQL operation
+* **Client**: executes operations against a GraphQL server endpoint, reusable and thread-safe 
+* **Response**: a holder for the GraphQL response including both data (which can be partial) and GraphQL errors if any
+* **InputType**: enables to build complex input objects.
+
+== Code examples
+
+=== Building a basic query
+
+In this example, we want to get the name of all heroes. The GraphQL query that fits that needs is:
+[source,json]
+----
+allHeroes {
+	name
+}
+----
+The way to build this query with the API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.QUERY)
+  .addRootField(new Field("allHeroes", 
+    new ArrayList<>(asList(new Field("name")))));
+----
+To be more concise and expressive, we also propose a lighter syntax based on predefined static methods:
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
```

Comment:
Yes, I will take inspiration from the JAX-RS counter-part:
 `Client client = ClientBuilder.newClient()`

So it will be something like that: 
`GraphQLBuilder builder = GraphQLBuilder.newBuilder(QUERY)`

Reply:
That's much better... a `GraphQLBuilder.newQuery()` would be even more compact.